tests/core/data_structures/test_networks.py

```
@@ -139,6 +142,75 @@ def _heisenberg_dense(L: int, Jx: float, Jy: float, Jz: float, h: float) -> np.n
     return H
 
 
+def _bose_hubbard_dense(length: int, local_dim: int, omega: float, J: float, U: float) -> np.ndarray:
+    """Construct the exact dense Bose–Hubbard Hamiltonian for comparison."""
```

Ruff (0.14.14)

[warning] Docstring contains ambiguous `–` (EN DASH). Did you mean `-` (HYPHEN-MINUS)?

(RUF002)